azure-pipelines.yml

```diff
@@ -10,151 +10,162 @@ resources:
+              export PATH=/usr/lib/llvm-8/bin:$PATH
+              export CC=clang
+              export CXX=clang++
+            displayName: 'Set up clang toolchain'
+          - script: |
+              bazel build --fat_apk_cpu=x86 //:android_dist
+            displayName: 'Build envoy.aar distributable'
+          - task: PublishPipelineArtifact@0
+            displayName: 'Publish envoy.aar distributable'
+            inputs:
+              artifactName: 'envoy.aar'
+              targetPath: 'dist/envoy.aar'
+      - job: linux_java_helloworld
+        dependsOn: linux_dist
+        timeoutInMinutes: 60
+        pool:
+          vmImage: 'Ubuntu 16.04'
+        variables:
+          android_ndk_home: /usr/local/lib/android/sdk/ndk-bundle
+        steps:
+          - checkout: self
+            submodules: true
+          - script: ./ci/linux_ci_setup.sh
+            displayName: 'Install dependencies'
+          - script: mkdir -p dist/
+            displayName: 'Create directory for distributable'
+          - task: DownloadPipelineArtifact@0
+            displayName: 'Download envoy.aar distributable'
+            inputs:
+              artifactName: 'envoy.aar'
+              targetPath: dist
+          - script: bazel build --fat_apk_cpu=x86 //examples/java/hello_world:hello_envoy
+            displayName: 'Build java app'
+      - job: linux_kotlin_helloworld
+        dependsOn: linux_dist
+        timeoutInMinutes: 60
+        pool:
+          vmImage: 'Ubuntu 16.04'
+        variables:
+          android_ndk_home: /usr/local/lib/android/sdk/ndk-bundle
+        steps:
+          - checkout: self
+            submodules: true
+          - script: ./ci/linux_ci_setup.sh
+            displayName: 'Install dependencies'
+          - script: mkdir -p dist/
+            displayName: 'Create directory for distributable'
+          - task: DownloadPipelineArtifact@0
+            displayName: 'Download envoy.aar distributable'
+            inputs:
+              artifactName: 'envoy.aar'
+              targetPath: dist
+          - script: bazel build --fat_apk_cpu=x86 //examples/kotlin/hello_world:hello_envoy_kt
+            displayName: 'Build kotlin app'
+  - stage: ios
+    dependsOn: [] # this removes the implicit dependency on previous stage and causes this to run in parallel.
+    jobs:
+      - job: dist
+        timeoutInMinutes: 60
+        pool:
+          vmImage: 'macos-10.14'
+        steps:
+          - checkout: self
+            submodules: true
+          - script: ./ci/mac_ci_setup.sh
+            displayName: 'Install dependencies'
+          - script: bazel build --config=ios //:ios_dist
+            displayName: 'Build Envoy.framework distributable'
+          - task: PublishPipelineArtifact@0
+            displayName: 'Publish Envoy.framework distributable'
+            inputs:
+              artifactName: 'Envoy.framework'
+              targetPath: 'dist/Envoy.framework'
+      - job: objc_helloworld
```

Comment:
Do you think we should prefix these with `mac` like we do for `linux` above?

Reply:
fair point, would be more homogenous